rotkehlchen/externalapis/routescan.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from typing import TYPE_CHECKING, Final
+
+from rotkehlchen.errors.misc import ChainNotSupported
+from rotkehlchen.externalapis.etherscan_like import EtherscanLikeApi
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import SUPPORTED_CHAIN_IDS, ApiKey, ChainID, ExternalService
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+ROUTESCAN_PAGINATION_LIMIT: Final = 10000
+ROUTESCAN_BASE_URL: Final = 'https://api.routescan.io/v2/network/mainnet/evm/{chain_id}/etherscan/api'
+ROUTESCAN_SUPPORTED_CHAINS: Final = (ChainID.ETHEREUM, ChainID.OPTIMISM)
+
+
+class Routescan(EtherscanLikeApi):
+    def __init__(
+            self,
+            database: 'DBHandler',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            database=database,
+            msg_aggregator=msg_aggregator,
+            service_name=ExternalService.ROUTESCAN,
+            pagination_limit=ROUTESCAN_PAGINATION_LIMIT,
+            default_api_key=ApiKey('placeholder'),  # as mentioned in their docs https://routescan.io/documentation
+        )
```

Comment:
if a user has an api key it will be able to add it right?


```suggestion
        super().__init__(
            database=database,
            msg_aggregator=msg_aggregator,
            service_name=ExternalService.ROUTESCAN,
            pagination_limit=ROUTESCAN_PAGINATION_LIMIT,
            default_api_key=ApiKey('placeholder'),  # placeholder is the string valid in the free tier as mentioned in their docs https://routescan.io/documentation
        )
```

Reply:
yeah, in `EtherscanLikeApi._query` it checks the api key from the db before using the default api key (same logic is used for etherscan).